### PR TITLE
Implement Serialize for VCObjectHash + provide golden tests for it

### PR DESCRIPTION
**Pull request by mirko-plowtech**
## Related issue
https://github.com/plow-technologies/inferno/issues/29

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -22,3 +22,4 @@ cabal.project.local~
 .HTF/
 .ghc.environment.*
 result*
+.vscode/
```
Comment (mirko-plowtech) [inline .gitignore:25]: minor improvement (outside the scope of the issue been addressed)

Comment (siddharth-krishna) [reply .gitignore:25]: Ooh, could we also add `.venv`? I'm working on some Python stuff for Inferno but it's going to take a while before I can open a PR, and my vscode goes nuts every time I switch back to the master branch 😅

Comment (mirko-plowtech) [reply .gitignore:25]: sure!

File: inferno-core/CHANGELOG.md [hunk 2]
```diff
@@ -1,9 +1,12 @@
 # Revision History for inferno-core
 
-## 0.1.0.4 -- 2022-01-13
+## 0.1.0.5 -- 2023-02-07
+* Add hspec-golden-cereal tests for VCObjectHash
+
+## 0.1.0.4 -- 2023-01-13
```
Comment (mirko-plowtech) [inline inferno-core/CHANGELOG.md:6]: some typos

File: inferno-types/src/Inferno/Types/VersionControl.hs [hunk 3]
```diff
@@ -94,6 +103,16 @@ instance Hashable VCObjectHash where
   hashWithSalt salt (VCObjectHash digest) =
     hashWithSalt salt $ convert @(Digest SHA256) @ByteString digest
 
+instance Serialize VCObjectHash where
+  get =
+    (getByteString 44)
```
Comment (mirko-plowtech) [inline inferno-types/src/Inferno/Types/VersionControl.hs:108]: SHA256 hashes have 256 bits of output, which translates to 64 characters in hex. That's double the length of the 32-character MD5 hash in hex. Base64 encoded SHA256 characters are only 44 bytes in length (and the last = for padding is omitted anyway).

File: inferno-types/src/Inferno/Types/VersionControl.hs [hunk 4]
```diff
@@ -282,12 +301,13 @@ pinnedUnderVCToMaybe = \case
   _ -> Nothing
 
 vcObjectHashToByteString :: VCObjectHash -> ByteString
```
Comment (mirko-plowtech) [inline inferno-types/src/Inferno/Types/VersionControl.hs:303]: We could remove `vcObjectHashToByteString` and `byteStringToVCObjectHash` but that would require to update `plow/all` code that relies on those functions too.

**Review by siddharth-krishna — APPROVED** [review]
LGTM thanks

**Pull request merged**